signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
+					return PolicyRequirementError(fmt.Sprintf("Signature for identity %s is not accepted", ref))
+				}
+				return nil
```

Comment:
Hum, this is awkward. If `signedIdentity` is `exact`Something (i.e. overrides the repo name being matched), we would want to reuse this in the pull-by-digest case as well so that the user only needs to specify the name override once.